source/transfer_list.rst

```diff
@@ -902,5 +902,42 @@ This manifest contains the SPMC attribute node consumed by the SPMD
      - hdr_size
      - Holds a SPMC manifest image in DT format.
 
+**DT Trusted Boot Firmware Configuration entry layout
```

Comment:
Would it be possible to put this information into the control dtb?

Reply:
No I don't think it would, the information in this dtb ( i.e. chain-of-trust, and mbedtls settings) is only intended for consumption by the trusted bootloader. Putting this information in the hardware dtb would require a client to strip this information prior to handover to later stages.

Reply:
Why is that? Where are the bindings for these nodes/properties?

Reply:
The nodes/properties are used specifically for the configuration of firmware. They aren't intended for use by any other components, and are unrelated to hardware descriptions. In some cases, the nodes are meant to be secret. Here is an example of the chain-of-trust bindings: https://review.trustedfirmware.org/plugins/gitiles/TF-A/trusted-firmware-a/+/refs/heads/master/docs/components/cot-binding.rst.

Reply:
> The nodes/properties are used specifically for the configuration of firmware. They aren't intended for use by any other components, and are unrelated to hardware descriptions. In some cases, the nodes are meant to be secret. Here is an example of the chain-of-trust bindings: https://review.trustedfirmware.org/plugins/gitiles/TF-A/trusted-firmware-a/+/refs/heads/master/docs/components/cot-binding.rst.

Thank you for the info. The binding looks like a good start but shouldn't it be in yaml format? How is this validated? I don't see mention of secret nodes. We don't have a way (at present) of specifying when nodes should be deleted before the next boot phase is invoked.

I see that the widevine binding is in the 'other' DT. This will get very confusing.

In general this seems to be a can of worms. We should have one DT schema and work to get it upstream

Reply:
> Each FW image may have its own FW_CONFIG contain data that only of relevance to its receiver and not subsequent components. This data is orthogonal to the HW device tree and device tree format is simply used as a convenient format for this data. This is all internal TF-A design stuff and doesn't need to be specified more widely.

This seems to open a distinction between the devicetree used by the system as a whole and the one used by TF-A. I have been pushing back against attempts to have U-Boot have its own DT. If TF-A has this, then we will need U-Boot to access two different DTs, won't we?

Why have a separate one? I mean, what is the benefit?

Reply:
> This seems to open a distinction between the devicetree used by the system as a whole and the one used by TF-A. 

TF-A consumes both system device tree and this special device tree
This TE is for Trusted Boot Configuration data which is consumed by BL2 (image loader) phase of TF-A. To load images with chain of trust we need image UUIDs and for some images (like Secure partitions) we also need load addresses. Considering we have a lot of images (32 standard images + Any number of SP's) we use device tree as a configuration file. 

> If TF-A has this, then we will need U-Boot to access two different DTs, won't we?

Reply:
Thanks for the responses.

I would like to see the lifetime of this information defined...e.g. you suggest that some pieces must not be passed onto the OS. Which ones and why not? It should be written out explicitly.

Also if we want to use a single DT, is that possible / allowed?

I also wish to note that you do not intend (I believe) to get these bindings accepted in dt-schema.

I suspect you plan to merge this anyway, but at least you have my objection noted. This will be the second PR merged over my objections.

Reply:
Hi Simon,

I've created a binding document within the TF-A documentation that outlines the usage of the TB_FW_CONFIG format. [This document is referenced in the specification and contains all the essential information](https://trustedfirmware-a.readthedocs.io/en/latest/components/fconf/tb_fw_bindings.html). Let us know if you're satisfied with this and we can look to get the change merged.

Reply:
Hi again,

Thank you for the binding. It helps a lot!

The properties should have hyphens, not underscores. It isn't clear to me why UUIDs are in there. The binding documentation is mostly not enough (for me) to figure out what it all means. For one thing, I don't see any mention of nodes.

Anyway the purpose of this PR is not to agree the binding. It just seems clear to me that the binding as is would need changes for dt-schema. Other projects have gone to enormous efforts to get bindings agreed (e.g. U-Boot). Could TF-A put in some effort to?

Could you clarify the lifetime of this information, as I requested?

Overall I believe it would be better to get an official binding agreed.

Reply:
> Hi again,
> 
> Thank you for the binding. It helps a lot!
> 
Thanks for your review!
> The properties should have hyphens, not underscores. It isn't clear to me why UUIDs are in there. The binding documentation is mostly not enough (for me) to figure out what it all means. For one thing, I don't see any mention of nodes.
> 
We currently are looking at decoupling individual properties in this dt to be created as seperate TE (https://github.com/FirmwareHandoff/firmware_handoff/pull/45 for mbetls_heap). Also, doing some experiments to see if we can live without UUIDs
> Anyway the purpose of this PR is not to agree the binding. It just seems clear to me that the binding as is would need changes for dt-schema. Other projects have gone to enormous efforts to get bindings agreed (e.g. U-Boot). Could TF-A put in some effort to?
> 
One of the purpose of Firmware handoff was to stop usage of device trees as Configuration files and we certainly will reduce dt usage in TF-A. Our plan is to reduce number of dt properties present in TF-A and for those which we can't remove, will put efforts to make the bindings in dt-schema.
> Could you clarify the lifetime of this information, as I requested?
> 
By lifetime if you mean the stage which would consume this information then it is BL2.
> Overall I believe it would be better to get an official binding agreed.
Ack